test/junit/scala/collection/immutable/TreeSetTest.scala

```diff
@@ -213,4 +213,33 @@ class TreeSetTest extends AllocationTest{
       assertEquals(s"$rText $lText", rhs, lhs)
     }
   }
+
+  @Test
+  def unionAndIntersectRetainLeft(): Unit = {
+    case class C(a: Int)(override val toString: String)
+    implicit val ordering: Ordering[C] = Ordering.by(_.a)
+    val c0l = C(0)("l")
+    val c0r = C(0)("r")
+    def assertIdenticalElements(expected: Set[C], actual: Set[C]): Unit = {
+      val expected1, actual1 = new java.util.IdentityHashMap[C, Unit]()
+      expected.foreach(expected1.put(_, ()))
+      actual.foreach(actual1.put(_, ()))
+      assertEquals(expected1, actual1)
+    }
+    assertIdenticalElements(Set(c0l), HashSet(c0l).union(HashSet(c0r)))
```

Comment:
We don't test +, which should also be consistent.

Reply:
Added.